tests/models/test_ModelUtils.py

```diff
@@ -6,22 +6,113 @@
 import pytest
 import torch
 import torch.nn as nn
+from packaging import version
 from pytest_mock import MockerFixture
 from torch import Tensor
 from torch.nn import Identity, Parameter
 
 from lightly.models import utils
 from lightly.models.utils import (
+    _mask_reduce,
+    _mask_reduce_batched,
     _no_grad_trunc_normal,
     activate_requires_grad,
     batch_shuffle,
     batch_unshuffle,
     deactivate_requires_grad,
+    masked_pooling,
     nearest_neighbors,
     normalize_weight,
     update_momentum,
 )
 
+scatter_min_torch_version = version.parse("1.12.0")
+torch_version = version.parse(torch.__version__)
+
+
+@pytest.mark.skipif(
+    torch_version < scatter_min_torch_version,
+    reason="scatter operations require torch >= 1.12.0",
+)
+class TestMaskReduce:
+    @pytest.fixture()
+    def mask1(self) -> Tensor:
+        return torch.tensor([[0, 0], [1, 2]], dtype=torch.int64)
+
+    @pytest.fixture()
+    def mask2(self) -> Tensor:
+        return torch.tensor([[1, 0], [0, 1]], dtype=torch.int64)
+
+    @pytest.fixture()
+    def feature_map1(self) -> Tensor:
+        feature_map = torch.tensor(
+            [[[0, 1], [2, 3]], [[4, 5], [6, 7]], [[8, 9], [10, 11]]],
+            dtype=torch.float32,
+        )  # (C H W) = (3, 2, 2)
+        return feature_map
+
+    @pytest.fixture()
+    def feature_map2(self) -> Tensor:
+        feature_map = torch.tensor(
+            [[[1, 2], [3, 4]], [[5, 6], [7, 8]], [[9, 10], [11, 12]]],
+            dtype=torch.float32,
+        )  # (C H W) = (3, 2, 2)
+        return feature_map
+
+    @pytest.fixture()
+    def expected_result1(self) -> Tensor:
+        res = torch.tensor(
+            [[0.5, 2.0, 3.0], [4.5, 6.0, 7.0], [8.5, 10.0, 11.0]], dtype=torch.float32
+        )
+        return res
+
+    @pytest.fixture()
+    def expected_result2(self) -> Tensor:
+        res = torch.tensor(
+            [[2.5, 2.5, 0.0], [6.5, 6.5, 0.0], [10.5, 10.5, 0.0]], dtype=torch.float32
+        )
+        return res
+
+    @pytest.mark.parametrize(
+        "feature_map, mask, expected_result",
+        [
+            ("feature_map1", "mask1", "expected_result1"),
+            ("feature_map2", "mask2", "expected_result2"),
+        ],
+    )
+    def test__mask_reduce(self, feature_map, mask, expected_result, request) -> None:
```

Comment:
Please also add types in tests, it helps understanding the tests

Reply:
fixed in https://github.com/lightly-ai/lightly/pull/1739/commits/9a631d84fd0caf4e67d5dc9c841292c7950a4a2b